File: C#/0242-valid-anagram.cs
// https://leetcode.com/problems/valid-anagram/description/

public class Solution {
    public bool IsAnagram(string s, string t) {
        Dictionary<char, int> s_dict = new Dictionary<char, int>();

        foreach (char character in s) {
            if (s_dict.ContainsKey(character)) {
                s_dict[character] += 1;
            } else {
                s_dict[character] = 1;
            }
        }

        foreach (char character in t) {
            if (s_dict.ContainsKey(character) && s_dict[character] > 0) {
                s_dict[character] -= 1;
            } else {
                return false;
            }
        }

        foreach (char character in s) {
            if (s_dict[character] > 0){
                return false;
            }
        }

        return true;
    }
}

File: Python3/0242-valid-anagram.py
# https://leetcode.com/problems/valid-anagram/

class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        s_dict = {}
        t_dict = {}
        
        for letter in s:
            if not letter in s_dict:
                s_dict[letter] = 1
            else:
                s_dict[letter] += 1
                
        for letter in t:
            if not letter in s_dict or s_dict[letter] <= 0:
                return False
            else:
                s_dict[letter] -= 1
        
        for key in s_dict:
            if s_dict[key] > 0:
                return False
        
        return True
